### PR TITLE
Toast component refactor

### PR DESCRIPTION
**Pull request by minhtuev**
## What changes are proposed in this pull request?

Refactor the Toast component so that it is more reusable and expose internal visibility state logic

## How is this patch tested? If it is not, please explain why.

* Manual testing (✅ )

Example usage:
```
export default function App() {
  return (
    <div className="App">
      <RecoilRoot>
        <ComponentA />
        <ComponentB />
        <Toast
          message="This is a notification"
          primary={(setOpen) => (
            <Button
              variant="contained"
              size="small"
              onClick={() => alert("Index created!")} // Send out an alert
              style={{ marginLeft: "auto" }} // Right align the button
            >
              Create an index
            </Button>
          )}
          secondary={(setOpen) => (
            <Button
              variant="text"
              color="secondary"
              size="small"
              onClick={(e) => someHandleClose(e, null, setOpen)} // Now closes the toast
              style={{ marginLeft: "auto" }} // Right align the button
            >
              Don't show me again
            </Button>
          )}
        />
      </RecoilRoot>
    </div>
  );
}
```
Playground sandbox: https://codesandbox.io/p/sandbox/example-recoil-forked-ywdpwn?workspaceId=9c715393-f638-4a9f-a4c5-cb7c71cd6f8c

See here for demo: https://www.loom.com/share/cabebf3fb8a04122a0d2ecd9966cf254?sid=2648bf0f-bd1f-4fb1-80d1-cdcadffe17b6

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Enhanced `Toast` component to accept custom button components (`primary` and `secondary`).
	- Transitioned state management to Recoil for improved flexibility.

- **Bug Fixes**
	- Updated export statements for the `Toast` component for consistency and clarity.

- **Documentation**
	- Updated interface and method signatures to reflect changes in the `Toast` component.

### CODE REVIEW

File: app/packages/components/src/components/Toast/Toast.tsx [hunk 1]
```diff
@@ -6,13 +1,22 @@
+import React from "react";
+import { atom, useRecoilState } from "recoil";
+import { Button, Snackbar, SnackbarContent } from "@mui/material";
 
 // Define types for the props
 interface ToastProps {
-  action: React.ReactNode;   // Accepts any valid React component, element, or JSX
   message: React.ReactNode;  // Accepts any valid React component, element, or JSX
+  primary: typeof Button;           // Accepts a Button component
+  secondary: typeof Button;          // Accepts a Button component
```
[the remark this answers is not in the document]
Comment (minhtuev) [reply app/packages/components/src/components/Toast/Toast.tsx]: ✅

File: app/packages/components/src/components/Toast/Toast.tsx [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
   duration?: number;         // Optional duration, with a default value
 }
 
-const Toast: React.FC<ToastProps> = ({ action, message, duration = 5000 }) => {
-  const [open, setOpen] = useState(true);
+const Toast: React.FC<ToastProps> = ({message, primary, secondary, duration = 5000 }) => {
+  const toastStateAtom = atom({
+    key: "open",
+    default: true,
+  });
```
[the remark this answers is not in the document]
Comment (minhtuev) [reply app/packages/components/src/components/Toast/Toast.tsx]: ✅

File: app/packages/components/src/components/Toast/Toast.tsx [hunk 3]
```diff
@@ -1,26 +1,43 @@
...
+  message: React.ReactNode;
+  primary: (setOpen: React.Dispatch<React.SetStateAction<boolean>>) => React.ReactNode;
+  secondary: (setOpen: React.Dispatch<React.SetStateAction<boolean>>) => React.ReactNode;
   duration?: number;         // Optional duration, with a default value
 }
 
-const Toast: React.FC<ToastProps> = ({ action, message, duration = 5000 }) => {
-  const [open, setOpen] = useState(true);
+const Toast: React.FC<ToastProps> = ({message, primary, secondary, duration = 5000 }) => {
+  const toastStateAtom = atom({
```
Comment (imanjra) [inline app/packages/components/src/components/Toast/Toast.tsx:14]: move this to state package or out of this component (i.e. to line 12)

File: app/packages/components/src/components/Toast/Toast.tsx [hunk 4]
```diff
@@ -1,26 +1,43 @@
...
+    key: "toastOpenState",
+    default: true,
+  });
+
+  const [open, setOpen] = useRecoilState(toastStateAtom); // State management for toast visibility
 
   const handleClose = (event, reason) => {
+    if (event) {
```
Comment (imanjra) [inline app/packages/components/src/components/Toast/Toast.tsx:22]: ✂️

Comment (minhtuev) [reply app/packages/components/src/components/Toast/Toast.tsx:22]: Done ✅

File: app/packages/components/src/components/Toast/Toast.tsx [hunk 5]
```diff
@@ -1,26 +1,43 @@
...
   };
 
+  const action = (
+    <div>
+      <Box display="flex" justifyContent="flex-end">
+        {primary(setOpen)} {/* Pass setOpen to primary button */}
+        {secondary(setOpen)} {/* Pass setOpen to secondary button */}
+      </Box>
+    </div>
+  );
```
[the remark this answers is not in the document]
Comment (minhtuev) [reply app/packages/components/src/components/Toast/Toast.tsx:36]: Wrong suggestion

**Review by Br2850 — APPROVED** [review]
👍🏽

**Pull request merged**